### ISSUE TITLE
Support the Jupyter (IPython) R kernel

### ISSUE THREAD

**Issue by macd** [aaren/notedown#24]
Project Jupyter now supports a R kernel (https://github.com/IRkernel/IRkernel)
It would be nice to support this in translating Rmd to ipynb by removing the 
%%R magics in the cells and then  adding the R specific metadata and 
language_info such as

 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.1.3"
  }

**Comment by aaren** [comment]
Agreed, this would be good. I'd like to maintain compatibility with the `%%R` magic though.

Perhaps we could use the `--rmagic` flag to toggle the behavior, i.e. use the R kernel with

```
notedown input.Rmd --knit 
```

but use `%%R` with

```
notedown input.Rmd --knit --rmagic
```

**Comment by macd** [comment]
That sounds like a good plan.  Thanks!

Don

**Comment by sje30** [comment]
On a similar point, will it support Julia code embedded in markdown?

**Comment by aaren** [comment]
@sje30 if I implement this it will be for the general case. If there is a Jupyter kernel written for the language then it should work.

**Comment by aaren** [comment]
Todo list for this:

- [x] install R kernel on my machine for testing (http://irkernel.github.io/installation/)
- [ ] add a `--kernel` argument that defaults to python
- [ ] set the language metadata when creating an internal notebook
    - [ ] kernelspec
    - [ ] language_info
- [ ] check whether this kernel spec is used when using `--run`
- [ ] allow setting metadata in a markdown yaml header?

**Comment by sje30** [comment]
Great, let me know if you need help.  There is a Julia kernel for Jupyter.

**Comment by aaren** [comment]
I'm probably not going to get around to this in the next couple of months unless I do some good procrastinating. Feel free to make a pull request in the meantime. The above list would be a good guideline.
